### PR TITLE
Inform user of sync errors

### LINKED ISSUE #61: Inform user of Sync/FxA errors
Child of #38 

If the user is logged in to Sync, there are two edge cases we might bump into:

1) their email is unverified
2) their password was changed on another device, so the locally-cached password is stale, and they need to sign in again to sync

In both cases, the existing desktop code redirects the user to the sync preferences page (about:preferences#sync) to fix their situation.

## Acceptance criteria
* Should redirect to the same endpoint
* Should alert the user when the email is unverified
* Should alert the user when the Account password changed
* Should define own entrypoint in telemetry

Open questions:
- Product: Do we want to just redirect to that same endpoint? Assuming yes, more questions:
- Eng: We should be able to open the correct link via either `gSync.openPrefs` or `gSync.openSignInAgainPage()` ([usage example](https://searchfox.org/mozilla-central/rev/c21d6620d384dfb13ede6054015da05a6353b899/browser/base/content/browser-sync.js#279-283)) from an embedded API experiment. Need to make sure `gSync` is available in an API experiment js file (it should be).
- UX: What kind of UX do we want to offer in Lockbox for those cases? Some kind of info screen and a button to redirect to the preferences page? Show an alert badge over the icon?
- Telemetry question: The existing desktop hamburger menu item appends a query string when sending users to the sync preferences page: "about:preferences?entrypoint=menupanel#sync". Do we need to register a new entrypoint? (markh's thinking was that this query string is part of fxa server telemetry, ingested via amplitude, so, maybe it's dynamic and "entrypoint=lockbox" doesn't need to be explicitly registered anywhere)

### Visual Design

**InVision**
Full View: https://mozilla.invisionapp.com/share/ENOL84959AK#/343774936_Logins_-_Sync_Issue_-Light-
Full View - User Menu: https://mozilla.invisionapp.com/share/ENOL84959AK#/343989952_Logins_-_Sync_Issue_-_Menu_-Light-
Doorhanger: https://mozilla.invisionapp.com/share/ENOL84959AK#/343777425_Doorhanger_-_Sync_Issue_-Light-

**Zeplin**
Full View: https://zpl.io/2ZjOwLJ
Full View - User Menu: https://zpl.io/adrod9e
Doorhanger: https://zpl.io/VOKELQl

*Link would direct to sync preferences

Requires #92

### PR DESCRIPTION
**Pull request by meandavejustice**
Fixes #61 

## Testing and Review Notes

You may test by opening up accounts.firefox.com in a private window and changing your password.

## Screenshots
popup-error-sync

### CODE REVIEW

**Comment by meandavejustice** [comment]
Just noticed the weirdness of the top border of the notification in the popup, it's abt 1 pixel off in the center section. Gunna investigate that and update before removing the "WIP", should be good for review otherwise.

**Comment by meandavejustice** [comment]
Inkedfixed-pop_LI

**Comment by meandavejustice** [comment]
@linuxwolf I think I've addressed your comments in https://github.com/mozilla-lockbox/lockbox-addon/pull/111/commits/04885bf17ac77386072b8e20c11e3819cedb095f, I renamed `profileReducer` => `profileWrap` in order to avoid a confusing structure like: `app.profile.profile`

File: src/list/manage/containers/app-header.js [hunk 1]
```diff
@@ -218,10 +218,10 @@ export class AppHeader extends React.Component {
 export default connect(
   ({
     app: {
-      tabsReducer: {
+      tabs: {
         selectedTab,
       },
-      profileReducer: {
+      profileWrap: {
```
Comment (linuxwolf) [inline src/list/manage/containers/app-header.js]: this makes sense

**Comment by changecourse** [comment]
Happy to suggest that we shorten to just "Reconnect" if we think this line break would be experienced by a decent percentage of users.

**Comment by linuxwolf** [comment]
> Happy to suggest that we shorten to just "Reconnect" if we think this line break would be experienced by a decent percentage of users.

while very anecdotal, the profile I used is a dev/testing one -- the only changes I deliberately made after creating it are Sync- and logins-related.  It might that a decent number of our Mac users will see it this way.

**Comment by meandavejustice** [comment]
>he button wraps to another line in the doorhanger:
ah I didn't see that on my machine. I could switch the font-size to use rem instead of px

**Pull request merged**